### ISSUE TITLE
ARG_TYPE_NONE passing non-zero value

### ISSUE THREAD

**Issue by bakkeby** [mihirlad55/dwm-ipc#14]
This is perhaps not a good example, but for some functions I allow for an optional client to allow functions to be more flexible than being restricted to only handling the currently selected client.
```c
void
zoom(const Arg *arg)
{
    Client *c = selmon->sel;
    if (arg && arg->v)
        c = (Client*)arg->v;
    if (!c)
        return;
...
```

In the configuration I would have the IPC command set up like this:
```c
IPCCOMMAND( zoom, 1, {ARG_TYPE_NONE} ),
```
which should just be passing in a 0 for the value, same as the keybinding.

Now when running this command via the IPC patch / dwm-msg I found dwm to be crashing, and I tracked it down to that a value was being passed in for this function (a client that doesn't exist).

I do not know if this has something to do with floating point representation or not, but changing the value from a float to an int (long) here seemed to fix this for me.
```diff
-     (*args)[0].f = 0;
+     (*args)[0].i = 0;
```
https://github.com/mihirlad55/dwm-ipc/blob/651ba97f111fdaec052dcf0e93a2a905b65e6922/ipc.c#L365

**Comment by bakkeby** [comment]
NB: I suspect that this has to do with that the float data type is 4 bytes in C whereas a long is 8 bytes. An Arg is a union which means that the different data types are stored in the same memory location. When memory is allocated I'd assume 8 bytes are reserved, but when setting the float value to 0 only the first 32 bits are cleared (set to 0) leaving the upper bytes with random data.

**Comment by mihirlad55** [comment]
Sorry for the late reply, I've been busy with school. Thank you for identifying the bug and a fix, I've released IPC Patch v1.5.6 with your fix.
